app/controllers/api/v2/smart_proxies_controller.rb

```diff
@@ -65,6 +65,21 @@ def refresh
         process_response @smart_proxy.refresh.blank? && @smart_proxy.save && @smart_proxy.reload
       end
 
+      api :POST, "/smart_proxies/:id/import_subnets", N_("Import subnets from Smart proxy")
+      param :id, String, :required => true
+
+      def import_subnets
+        proxy = SmartProxy.find(params[:id])
+        @subnets = Subnet::Ipv4.import(proxy)
+        if @subnets.nil?
+          render_message _('Proxy does not have a DHCP feature'), :status => :unprocessable_entity
+        elsif @subnets.empty?
+          render_message _('No new subnets were found'), :status => :no_content
+        else
+          render_message _('New subnets were found and imported'), :status => :created
```

Comment:
It's quite common in REST to actually return the created results rather than a message. I'd expect to see that. In the case of no content it could return an empty array.

AFAIK the UI has no way of showing which subnets were imported, the API could actually be a good way to do so. This also allows the client to do some things like counting how many new subnets. Perhaps even showing a small summary.

`render_message` is mostly used to show errors.

Reply:
@ekohl thank you for the clarification , I'll change that :)